src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkMissingWildcardsInspection.kt

```diff
@@ -0,0 +1,51 @@
+package com.jetbrains.snakecharm.inspections
+
+import com.intellij.codeInspection.LocalInspectionToolSession
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.psi.util.PsiTreeUtil
+import com.jetbrains.snakecharm.SnakemakeBundle
+import com.jetbrains.snakecharm.lang.SnakemakeNames
+import com.jetbrains.snakecharm.lang.psi.SmkRuleOrCheckpoint
+import com.jetbrains.snakecharm.lang.psi.SmkRuleOrCheckpointArgsSection
+import com.jetbrains.snakecharm.lang.psi.SmkWildcardsCollector
+
+class SmkMissingWildcardsInspection : SnakemakeInspection() {
+    override fun buildVisitor(
+            holder: ProblemsHolder,
+            isOnTheFly: Boolean,
+            session: LocalInspectionToolSession
+    ) =
+            object : SmkWildcardsInspectionVisitor<SmkRuleOrCheckpointArgsSection>(holder, session) {
+        override fun visitSmkRuleOrCheckpointArgsSection(st: SmkRuleOrCheckpointArgsSection) {
+            if (st.name == SnakemakeNames.SECTION_OUTPUT ||
+                    st.name !in SmkRuleOrCheckpointArgsSection.KEYWORDS_GENERATING_WILDCARDS) {
+                return
+            }
+
+            updateDeclarationAndWildcards(
```

Comment:
Easy to forget to call this method and get an incorrect result, API not good here